nautobot/dcim/api/serializers.py

```diff
@@ -1120,6 +1122,20 @@ class Meta:
         ]
 
 
+class RedundancyGroupSerializer(NautobotModelSerializer, TaggedObjectSerializer, StatusModelSerializerMixin):
+    url = serializers.HyperlinkedIdentityField(view_name="dcim-api:redundancygroup-detail")
+    failover_strategy = ChoiceField(choices=RedundancyGroupFailoverStrategyChoices)
```

Comment:
Should the serializer also have a nested serializer for the associated devices?

Reply:
@bryanculver @glennmatthews Will we only have *Device* redundancyGroups in the future? Can Interface / VirtualChassis in the future form its RedundancyGroup?

Shouldn't we rename this model to `DeviceRedundancyGroup` ?

Reply:
Hi @mzbroch one of the major goals here is to not boil the ocean and solve the problem in an agile sense.

The implementation in this PR is about modeling device chassis-level redundancy, nothing more. In doing so, it is trying to provide a data model specific to the needs of that use case.

#2621 was proposed recently as a related solution to hext-hop redundancy protocol modeling. It is similar, yes, but in many ways distinct from this use case. For instance, the ability for interfaces to participate in multiple hext-hop protocols, or the need to associate IP addresses.

My problem with a generic grouping feature for these use cases is that it is just that, generic. As an end user, I would hate to have to create a bunch of custom fields and relationships just to solve the redundancy use cases at the 80% level. We should be providing better data models & UX for those needs.

Let me be clear though, I _do_ see a need for a "static grouping" feature, just not to solve this particular use case.

So I do see us having several models related to redundancy, but targeted at their respective, specific networking use cases.